src/components/UriViewer/UriViewer.js

```diff
@@ -56,28 +64,70 @@ export const UriViewer = _.flow(
         setMetadata(metadata)
       }
     } catch (e) {
-      setLoadingError(await e.json())
+      // azure blob storage api only returns responses in xml format at this time
+      // https://feedback.azure.com/d365community/idea/0646f944-3a25-ec11-b6e6-000d3a4f0f84
+      if (isAzureUri(uri)) {
+        setLoadingError(await e.text())
+      } else {
+        setLoadingError(await e.json())
+      }
     }
   }
+
   useOnMount(() => {
     loadMetadata()
   })
 
   const { size, timeCreated, updated, bucket, name, fileName, accessUrl } = metadata || {}
   const gsUri = `gs://${bucket}/${name}`
   const downloadCommand = getDownloadCommand(fileName, gsUri, accessUrl)
-  if (isAzureUri(uri)) {
+
+  if (isAzureUri(uri) && azureStorage !== undefined) {
+    const { azureSasStorageUrl, fileName, lastModified, size } = azureStorage
+    uri = azureSasStorageUrl || uri
     return h(Modal, {
       onDismiss,
       title: 'File Details',
       showCancel: false,
       showX: true,
       okButton: 'Done'
-    },
-    [els.cell([
-      els.label('Filename'),
-      els.data(_.last(uri.split('/')).split('.').join('.\u200B')) // allow line break on periods
-    ]), div({ style: { marginTop: '2rem', fontSize: 14 } }, ['Download functionality for Azure files coming soon'])])
+    }, [
+      Utils.cond(
+        [loadingError, () => h(Fragment, [
+          div({ style: { paddingBottom: '1rem' } }, [
+            'Error loading data. This file does not exist or you do not have permission to view it.'
+          ])
+        ])],
+        [azureStorage, () => h(Fragment, [
+          els.cell([
+            els.label('Filename'),
+            els.data(fileName.split('.').join('.\u200B')) // allow line break on periods
+          ]),
+          azureStorage.lastModified && els.cell([
+            els.label('Updated'),
+            els.data(new Date(lastModified).toLocaleString())
+          ]),
+          els.cell([els.label('File size'), els.data(filesize(size))]),
+          h(AzureUriDownloadButton, { uri, fileName }),
+          els.cell([
+            els.label('Terminal download command'),
+            els.data([
+              div({ style: { display: 'flex' } }, [
+                input({
+                  readOnly: true,
+                  value: `azcopy copy '${azureSasStorageUrl}' .`,
+                  style: { flexGrow: 1, fontWeight: 400, fontFamily: 'Menlo, monospace' }
+                }),
+                h(ClipboardButton, {
+                  text: `azcopy copy '${azureSasStorageUrl}' .`,
+                  style: { margin: '0 1rem' }
```

Comment:
Please add an aria-label to this (and also in the other location you copied it from). Without the label, the screen reader just reads "button".

![image](https://user-images.githubusercontent.com/484484/228936777-7f5246ad-3d9b-4c99-ba5d-48f1c356396e.png)

Example:

```
h(ClipboardButton, {
            'aria-label': 'Copy SAS URL to clipboard',
            text: storageDetails.azureContainerSasUrl, style: { marginLeft: '0.25rem' }
          })